### PR TITLE
LPS-161807 Modification of value buttons from 'close' to 'cancel'

### PR DESCRIPTION
**Pull request by samuelleolopes**

### CODE REVIEW

**Comment by samuelleolopes** [comment]
## :fire: GitRay Report 
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
 ## Jira Tasks 

 <html><details><summary>More details about: <b>LPS-161807 [FE] Create Logic of Change the User Password</b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-161807">[FE] Create Logic of Change the User Password</a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Minor</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Progress</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Samuel Lopes</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Samuel Lopes</td>
        </tr>
                </table>
          </details><details><summary>More details about: <b>LPS-161803 [FE] Create Input Current Password </b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-161803">[FE] Create Input Current Password </a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Minor</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Progress</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Samuel Lopes</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Samuel Lopes</td>
        </tr>
                </table>
          </details></html> 

---

Hey, don't forget to update all related tickets with correct status, such as **In Review** :wink: 
### Git Stats 

7 files changed, 157 insertions(+), 61 deletions(-)

**Comment by samuelleolopes** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/pages/Manage/User/ChangeUserPassword.tsx [hunk 1]
```diff
@@ -14,25 +14,82 @@
...
 		formState: {errors},
 		handleSubmit,
 		register,
-	} = useForm({resolver: yupResolver(yupSchema.password)});
+		setError,
+	} = useForm({
+		defaultValues: userAccount,
+		reValidateMode: 'onSubmit',
+		resolver: yupResolver(
+			validationId ? yupSchema.passwordRequired : yupSchema.password
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/pages/Manage/User/ChangeUserPassword.tsx:62]:
```suggestion
			isSelfAccount ? yupSchema.passwordRequired : yupSchema.password
```

File: modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/pages/Manage/User/ChangeUserPassword.tsx [hunk 2]
```diff
@@ -43,33 +100,44 @@ const ChangeUserPassword: React.FC = () => {
 	return (
 		<Container className="change-user-password">
 			<ClayForm>
-				<div>
+				{validationId ? (
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/pages/Manage/User/ChangeUserPassword.tsx:103]:
```suggestion
				{isSelfAccount ? (
```

File: modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/pages/Manage/User/ChangeUserPassword.tsx [hunk 3]
```diff
@@ -43,33 +100,44 @@ const ChangeUserPassword: React.FC = () => {
...
 					<Form.Input
 						{...inputProps}
-						label={i18n.translate('new-password')}
-						name="password"
-						placeholder="Password"
+						label={i18n.translate('current-password')}
+						name="currentPassword"
+						placeholder={i18n.translate('current-password')}
+						required={true}
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/pages/Manage/User/ChangeUserPassword.tsx:109]:
```suggestion
						required
```

File: modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/pages/Manage/User/ChangeUserPassword.tsx [hunk 4]
```diff
@@ -14,25 +14,82 @@
...
+				update: (...params) =>
+					liferayUserAccountsRest.update(...params),
+			}
+		)
+			.then(mutatePassword)
+			.then(() => onSave())
+			.catch((error) => {
+				if (error.info.type === 'MustMatchCurrentPassword') {
+					return setError('currentPassword', {
+						message: i18n.translate('current-password-incorrect'),
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/pages/Manage/User/ChangeUserPassword.tsx:86]:
```suggestion
						message: i18n.translate('current-password-is-incorrect'),
```

File: modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/schema/yup.ts [hunk 5]
```diff
@@ -25,22 +25,45 @@ const userSchema = yup.object({
...
+	currentPassword: yup.string(),
+
+	password: yup
+		.string()
+		.required(i18n.translate('this-field-is-required'))
+		.min(8, i18n.translate('minimum-8-characters'))
+		.matches(
+			/[a-zA-Z0-9]/,
+			i18n.translate('password-may-contain-letters-and-number')
+		),
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/schema/yup.ts:37]:
```suggestion

const MINIMUM_CHARACTERS_PASSWORD = 8;

const passwordStructure = {
	currentPassword: yup.string(),

	password: yup
		.string()
		.required(i18n.translate('this-field-is-required'))
		.min(MINIMUM_CHARACTERS_PASSWORD, i18n.sub('minimum-x-characters', MINIMUM_CHARACTERS_PASSWORD.toString()))
		.matches(
			/[a-zA-Z0-9]/,
			i18n.translate('password-may-contain-letters-and-number')
		),
```

Comment (kevenleone) [reply modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/schema/yup.ts:37]: Avoid magic numbers

File: modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/services/rest/LiferayUserAccounts.ts [hunk 6]
```diff
@@ -13,33 +13,33 @@
...
-
-const createUserAccount = (user: User) =>
-	fetcher.post('/user-accounts', adapter(user));
-
-const updateUserAccount = (id: number, user: User) =>
-	fetcher.patch(`/user-accounts/${id}`, adapter(user as User));
-
-export {createUserAccount, updateUserAccount, getUserAccountQuery};
+const liferayUserAccountsRest = new LiferayUserAccountsRest();
+export {liferayUserAccountsRest};
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/extra/remote-app/src/services/rest/LiferayUserAccounts.ts:45]: Add blank space

```suggestion
const liferayUserAccountsRest = new LiferayUserAccountsRest();

export {liferayUserAccountsRest};
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
LGTM 🔥 

But see my suggestions, please

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**